webview-ui/src/components/common/MermaidBlock.tsx

```diff
@@ -86,45 +87,85 @@ interface MermaidBlockProps {
 	code: string
 }
 
-export default function MermaidBlock({ code }: MermaidBlockProps) {
+// kilocode_change next line rename to originalCode to keep the difference with Roo smaller
+export default function MermaidBlock({ code: originalCode }: MermaidBlockProps) {
 	const containerRef = useRef<HTMLDivElement>(null)
 	const [isLoading, setIsLoading] = useState(false)
 	const [error, setError] = useState<string | null>(null)
 	const [isErrorExpanded, setIsErrorExpanded] = useState(false)
+	// kilocode_change start
+	const [isFixing, setIsFixing] = useState(false)
+	const [code, setCurrentCode] = useState(originalCode)
+	const [hasAutoFixed, setHasAutoFixed] = useState(false)
+	// kilocode_change end
 	const { showCopyFeedback, copyWithFeedback } = useCopyToClipboard()
 	const { t } = useAppTranslation()
 
 	// 1) Whenever `code` changes, mark that we need to re-render a new chart
 	useEffect(() => {
 		setIsLoading(true)
 		setError(null)
-	}, [code])
+		// kilocode_change start
+		setCurrentCode(originalCode)
+		setHasAutoFixed(false)
+		// kilocode_change end
+	}, [originalCode])
 
-	// 2) Debounce the actual parse/render
+	// 2) Debounce the actual parse/render with LLM-powered fixing
```

Comment:
unnecessary change?

Reply:
Maybe? I did intentionally keep this because it is a bit unexpected that it does that when you're reading the code here. And when I make the PR to Roo I think this makes sense as well?